### DISCUSSION TITLE
Library to create and transmit wmbus telegrams

### DISCUSSION THREAD

**Discussion by asbachb** [wmbusmeters/wmbusmeters#1339] [General]
Hi. I'm aware that the topic might not match the project. But I'm wondering if there are any libs which helps to create and transmit wmbus telegrams.
So basically I want to build an esp based reader which propagates it's readings via wmbus.

/Cc @weetmuts

**Comment by weetmuts** [comment]
This is very interesting. Drop me an email and we can discuss.

**Comment by chris-est** [comment]
Hi, is there any progress on this? I would be interested in a library to create wmbus telegrams as well.

My use case is this:
A USB stick (iu891a) receives wmbus telegrams from my water-meter (working fine).
My gas-meter has no digital output but i can get the usage from a reed-switch and want to transmit that data using a wmbus module (WürthElektronik Metis-E).
--> So to summarize: I want to emulate a gas-meter wmbus packet (preferred: for a driver already present in wmbusmeters) to transmit <total_m³>.
